### ISSUE TITLE
Snowball cannon

### ISSUE THREAD

**Issue by JavidPack** [bluemagic123/tModLoader#132]
Snowball cannon will use the wrong projectile if a modded ammo is used (iirc)

**Comment by Jofairden** [comment]
Hmm, I tested this using the example bullet when setting ammo to AmmID.Snowball, it was working fine.

**Comment by JavidPack** [comment]
Hmm....I think the error might actually be Weapons using rocket 1,2,3,4 shoot the wrong thing. It was a while ago, but looking at the code something should be messing up.

**Comment by JavidPack** [comment]
Update: I just realized that I wrote Snowball Cannon not Snowman Cannon. Snowball Cannon uses rockets as ammunition.

Anyway, the problem is in PickAmmo there is this code:
```// ItemID.SnowmanCannon = 1946;
if (sItem.type == 1946)
{
	// ProjectileID.RocketSnowmanI = 338;
	// ItemID.RocketI = 771;
	shoot = 338 + item.type - 771;
}
else if (sItem.useAmmo == AmmoID.Rocket)
{
	shoot += item.shoot;
}```

Which causes problems. Modded rocket ammo can work if Grenade, Rocket, and ProximityMine projectile variants are made (and loaded in correct order) and the ammo's item.shoot is set to the Grenade projectile minus 133 (since 133, 134, or 135 is added to the ammos shoot by the various rockey launchers), but that math doesn't work for the snowman cannon since it has it's own 4 projectiles. 

I'm not sure what to do. I'll just disable modded rocket ammo for snowman cannon so people don't complain about weird projectiles (or errors) shooting out.
